src/Appwrite/Platform/Tasks/Backup.php

```diff
@@ -0,0 +1,439 @@
+            ->inject('log')
+            ->callback(fn(string $database, Group $pools, ?Logger $logger, Log $log) => $this->action($database, $pools, $logger, $log));
+    }
+
+    public static function getName(): string
+    {
+        return 'backup';
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function action(string $database, Group $pools, ?Logger $logger, Log $log): void
+    {
+        $this->checkEnvVariables();
+
+        $this->database = $database;
+        $this->dsn = $this->getDsn($database);
+        if (is_null($this->dsn)) {
+            throw new Exception('No DSN match');
+        }
+
+        //todo: remove this:
+        $info = str_replace($this->dsn->getPassword(), "*", 'Trying to connect to ' . $this->dsn->getHost() . ' : ' . $this->dsn->getPort() . ' user: ' . $this->dsn->getUser() . ' password: ' . $this->dsn->getPassword());
+        console::info($info);
+
+        $dsn = new DSN(App::getEnv('_APP_CONNECTIONS_BACKUPS_STORAGE', ''));
+        $this->s3 = new DOSpaces('/' . $database . '/' . self::VERSION, $dsn->getUser(), $dsn->getPassword(), $dsn->getPath(), $dsn->getParam('region'));
+
+        try {
+            $this->retry(function () use ($pools, $database) {
+                $pools
+                ->get('replica_' . $database)
+                ->pop()
+                ->getResource();
+            }, 10, 5);
+        } catch (Exception $error) {
+            throw new Exception('Failed to connect to database: ' . $error->getMessage());
+        }
+
+        $this->setContainerId();
+        $this->setProcessors();
+
+        Console::loop(function () use ($logger, $log) {
+            try {
+                $this->start();
+            } catch (Exception $error) {
+                if ($logger) {
+                    $log->setNamespace("appwrite-backup");
+                    $log->setType(Log::TYPE_ERROR);
+                    $log->setMessage($error->getMessage());
+                    $log->setAction('appwrite-backup');
+                    $log->addExtra('file', $error->getFile());
+                    $log->addExtra('line', $error->getLine());
+                    $log->addExtra('trace', $error->getTraceAsString());
+                    $log->addExtra('detailedTrace', $error->getTrace());
+
+                    $isProduction = App::getEnv('_APP_ENV', 'development') === 'production';
+                    $log->setEnvironment($isProduction ? Log::ENVIRONMENT_PRODUCTION : Log::ENVIRONMENT_STAGING);
+
+                    $responseCode = $logger->addLog($log);
+                    Console::info('Usage stats log pushed with status code: ' . $responseCode);
+                }
+
+                Console::error('[Error] Time: ' . date('Y-m-d H:i:s'));
+                Console::error('[Error] Message: ' . $error->getMessage());
+                Console::error('[Error] File: ' . $error->getFile());
+                Console::error('[Error] Line: ' . $error->getLine());
```

Comment:
Let's add the trace as a string here

Reply:
The logger always prints this out cli.php Line 179:
Console::warning("Failed: {$error->getMessage()}");
Console::warning($error->getTraceAsString());
So no need for the trace, the message will be printed as a warning as well, but I think we can print it as an error as well.
I will DM you a print screen .